plugin/src/main/java/com/denizenscript/denizen/scripts/commands/item/MapCommand.java

```diff
@@ -90,100 +87,84 @@ public MapCommand() {
         }
-        if (resetLoc == null && image == null && script == null && dot == null && text == null) {
+        if (resetLoc == null && image == null && script == null && dot == null && text == null && !resetEmpty) {
             throw new InvalidArgumentsRuntimeException("Must specify a valid action to perform!");
         }
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), id, create, resetLoc, image, script, dot, radius, scale, db("resize", resize), db("tracking", tracking), width, height, x, y, text);
-        }
         MapView map;
         if (create != null) {
             map = Bukkit.getServer().createMap(create.getWorld());
             scriptEntry.saveObject("created_map", new ElementTag(map.getId()));
             Debug.echoDebug(scriptEntry, "Created map with id " + map.getId() + ".");
         }
         else { // id != null
-            map = Bukkit.getServer().getMap((short) id.asInt());
+            map = Bukkit.getServer().getMap(id.asInt());
             if (map == null) {
                 Debug.echoError("No map found for ID '" + id.asInt() + "'!");
                 return;
             }
         }
-        if (resetLoc != null) {
+        if (resetEmpty || resetLoc != null) {
             map.setTrackingPosition(tracking);
             if (scale != null) {
-                map.setScale(MapView.Scale.valueOf(scale.asString().toUpperCase()));
+                map.setScale(scale);
             }
-            List<MapRenderer> oldRenderers = DenizenMapManager.removeDenizenRenderers(map);
-            for (MapRenderer renderer : oldRenderers) {
+            for (MapRenderer renderer : DenizenMapManager.removeDenizenRenderers(map)) {
                 map.addRenderer(renderer);
             }
-            map.setCenterX(resetLoc.getBlockX());
-            map.setCenterZ(resetLoc.getBlockZ());
-            map.setWorld(resetLoc.getWorld());
+            if (resetLoc != null) {
+                map.setCenterX(resetLoc.getBlockX());
+                map.setCenterZ(resetLoc.getBlockZ());
+                map.setWorld(resetLoc.getWorld());
+            }
         }
         if (script != null) {
             DenizenMapManager.removeDenizenRenderers(map);
             ((MapScriptContainer) script.getContainer()).applyTo(map);
         }
+        DenizenMapRenderer dmr = DenizenMapManager.getDenizenRenderer(map);
         if (image != null) {
-            DenizenMapRenderer dmr = DenizenMapManager.getDenizenRenderer(map);
-            int wide = width != null ? width.asInt() : resize ? 128 : 0;
-            int high = height != null ? height.asInt() : resize ? 128 : 0;
-            if (image.asLowerString().endsWith(".gif")) {
+            width = formatSize(width, resize);
+            height = formatSize(height, resize);
+            if (CoreUtilities.toLowerCase(image.asString()).endsWith(".gif")) {

```

Comment:
Why not `asLowerString` like before? (or alternatively can also make the arg type `String`)